src/k2/cmd/tpcc/transactions.h

```diff
@@ -800,3 +800,374 @@ class OrderStatusT : public TPCCTxn
+                query_oid.setLimit(1);
+                query_oid.setReverseDirection(false);
+
+                std::vector<String> projection{"OID"};  // make projection
+                query_oid.addProjection(projection);
+                dto::expression::Expression filter{};   // make filter expression
+                query_oid.setFilterExpression(std::move(filter));
+
+                return _txn.query(query_oid)
+                .then([this, idx, &query_oid] (auto&& response) mutable {
+                    if (!response.status.is2xxOK()) {
+                        K2DEBUG("DeliveryT getOrderID query response Error");
+                        _failed = true;
+                        return make_ready_future<bool>(false);
+                    }
+
+                    if (response.records.size()) {
+                        dto::SKVRecord& rec = response.records[0];
+                        // get order id
+                        std::optional<int64_t> orderIDOpt = rec.deserializeField<int64_t>("OID");
+                        _NO_O_ID[idx] = *orderIDOpt;
+
+                        K2DEBUG("DeliveryT parallel request " << idx << ": Get Order ID. w_id:" << _w_id << ", d_id:"
+                                << _d_id[idx] << ", order_id:" << _NO_O_ID[idx]);
+
+                        return make_ready_future<bool>(true);
+                    } else {
+                        K2DEBUG("DeliveryT parallel request " << idx << ": not found Order ID matches w_id:" 
+                                << _w_id << ", d_id:" << _d_id[idx]);
+                        // set all output values to -1
+                        _NO_O_ID[idx] = -1;
+                        _O_C_ID[idx] = -1;
+                        _OL_SUM_AMOUNT[idx] = -1;
+
+                        // report MISS_condition occurs more than once of the delivery transaction
+                        if (_miss_once) {
+                            K2INFO("TPCC DeliveryT MISS_Condition: occurs more than once! not Found Order ID matches w_id["
+                                    << _w_id << "] & d_id[" << _d_id[idx] << "] in New-Order table"); 
+                        }
+                        _miss_once = true;
+
+                        return make_ready_future<bool>(false);
+                    }
+                });
+            });
+        });
+    }
+
+    // delete the selected row in NEW-ORDER table
+    future<> delNewOrderRow(uint32_t idx) {
+        NewOrder delNORow(_w_id, _d_id[idx], _NO_O_ID[idx]);
+        return writeRow<NewOrder>(delNORow, _txn, true).discard_result();
+    }
+
+    // retrieve the "LineCount" and "CID" in ORDER table with matching _w_id, _d_id and _no_o_id
+    future<> getCustomerID(uint32_t idx) {
+        return _client.createQuery(tpccCollectionName, "order")
+        .then([this, idx] (auto&& response) mutable {
+            CHECK_READ_STATUS(response);
+
+            return do_with(std::move(response.query), [this, idx] (Query& query_cid) {
```

Comment:
This is a query instead of a read because you want to do projection? If so can you add a comment explaining that?